src/URIs.jl

```diff
@@ -453,31 +455,54 @@ normpath(url::URI) =
     URI(scheme=url.scheme, userinfo=url.userinfo, host=url.host, port=url.port,
         path=normpath(url.path), query=url.query, fragment=url.fragment)
 
+# normpath helper functions
+_tail(s, prefix) = last(s, length(s) - length(prefix))
+function _pop_segment(buf)
+    last_slash = findlast('/', buf)
+    (last_slash === nothing) ? "" : buf[begin:prevind(buf, last_slash)]
```

Comment:
The `buf[begin:...]` syntax isn't valid for 1.0 (hence the failing CI tests). I'd be fine bumping up the required Julia version to 1.6, since that's the new LTS, or I think we could change this to `buf[firstindex(buf):...]`.

Reply:
Sorry for the delay, I've gone ahead and updated the code to use `firstindex` instead of `begin` for backwards compatibility. But yeah, since 1.0 isn't supported anymore, I'm not sure how big an issue it is.